### DISCUSSION TITLE
remove ring, shadow, border from input

### DISCUSSION THREAD

**Discussion by kfbfarley** [tailwindlabs/tailwindcss#4007] [Help]
Hey Everyone,

I'm having such a tiny problem while trying to remove the focus ring, border, or shadow from the input.

What I have:

![image](https://user-images.githubusercontent.com/42237552/113945318-66643100-9806-11eb-80d4-76d5805e0c65.png)

What I want:

![image](https://user-images.githubusercontent.com/42237552/113945352-7714a700-9806-11eb-8761-682f5b45e32d.png)

The css class code: 

`ring-0 focus:ring-0 shadow-none focus:shadow-none focus:outline-none border-none focus:border-none h-14 w-full md:w-4/5`

If you could gimme a hand that would be amazing and much appreciated.

Thank you,
Farley.

**Comment by simonswiss** [comment]
Hey!

Are you using the `@tailwindcss/forms` plugin?

Are you able to put the HTML for this input in a https://play.tailwindcss.com example? That'll make it much easier to "fork" it and show you how to fix it 👍

**Reply by kfbfarley** [reply]
Hey Sim,

Thank you for your time in the first place.

It's right here:
https://play.tailwindcss.com/jrf1oYYioA

Congrats,
Farley.

**Reply by simonswiss** [reply]
I am a bit confused - looks like you are trying to _undo_ a lot of styles (borders, shadows, ring, etc).

Are you trying to use Tailwind CSS on top of another CSS library and override the styles?

See my other answer below for an example - I have simplified a bunch of things in your Tailwind config, to focus on what is relevant here.

**Comment by simonswiss** [comment]
Hey!

I have created a Tailwind Play with something like you are trying to achieve:

https://play.tailwindcss.com/OMiAPIoezU

Please note: 

It's not good for accessibility to completely remove the focus styles. For that reason, I also suggested another example with subtle, but visible focus styles 👍

![image](https://user-images.githubusercontent.com/485747/113955003-46f8f280-985e-11eb-89ed-2d9905e5d512.png)

**Reply by kfbfarley** [reply]
Thank you so much for your solution.

The problem is here:
`colors: {
      ...colors,
    },`

Do you have any idea why, and know if there's a solution?

I got your solution working but my previous styles are not in place as it was because I need _colors_ imports. 

Thanks in advance.

**Reply by kfbfarley** [reply]
Here is what is generated that is causing the unwanted CSS values

![image](https://user-images.githubusercontent.com/42237552/113958881-1dba7100-9822-11eb-9eec-21c70643cd18.png)

**Reply by kfbfarley** [reply]
Hey,

The solution is exactly what you said, using visible focus styles, but with the same background color.

`focus:ring-brand-yellow focus:border-brand-yellow focus:shadow-outline border-none`

Thank you in advance,
Farley.

**Comment by joshxyzhimself** [comment]
I did the following lol

```css
* {
  @apply appearance-none outline-none;
}
*:focus {
  @apply appearance-none outline-none;
}
```

**Comment by noman-zaag** [comment]
thank you..
